### PR TITLE
Use PKCE instead of traditional oauth flow

### PR DESCRIPTION
**Pull request by DRSchlaubi**

### CODE REVIEW

File: app/src/main/java/de/hbch/traewelling/api/PKCEUtil.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    fun Context.openOAuthAuthorizationPage() {
+        val intent = CustomTabsIntent.Builder()
+            .setShowTitle(false)
+            .build()
+
+        val (state, challenge) = nextChallenge()
+        val url = Uri.parse("https://traewelling.de").buildUpon()
+            .appendPath("oauth").appendEncodedPath("authorize")
+            .appendQueryParameter("client_id", "28")
```
Comment (jheubuch) [inline app/src/main/java/de/hbch/traewelling/api/PKCEUtil.kt:108]: This client id is hard coded?

**Review by jheubuch — CHANGES_REQUESTED** [review]

**Comment by jheubuch** [comment]
Maybe we also can store the refresh token so that we can refresh the tokens before they expire.
Also we should wait to merge this, until scopes are ready and then implement the scopes we need.

**Comment by DRSchlaubi** [comment]
Sounds reasonable, I will also base the push notifications PR on this as it requires modification of auth logic as well

File: app/src/main/java/de/hbch/traewelling/api/PKCEUtil.kt (hunk 1 above)
Comment (DRSchlaubi) [reply app/src/main/java/de/hbch/traewelling/api/PKCEUtil.kt:108]: yes it is, that was an oversight

**Pull request closed without merging**